app/javascript/app/pages/national-context/socioeconomic/population/population-selectors.js

```diff
@@ -0,0 +1,275 @@
+const getProvinceIndicatorsForPopulationOptions = createSelector(
+  [ getIndicators ],
+  indicators => {
+    if (!indicators) return null;
+
+    const POPULATION_INDICATOR = 'pop_total';
+    const options = [];
+
+    const populationProvinces = indicators &&
+      indicators.values &&
+      indicators.values.filter(
+        ind =>
+          ind.location_iso_code3 !== 'IDN' &&
+            ind.indicator_code === POPULATION_INDICATOR
+      );
+
+    if (populationProvinces) {
+      populationProvinces.forEach(
+        province =>
+          options.push({ label: province.location, value: province.location })
+      );
+    }
+
+    return sortBy(options, 'label');
+  }
+);
+
+const getFilterOptions = createStructuredSelector({
+  popNationalIndicator: getNationalIndicatorsForPopulationOptions,
+  popProvince: getProvinceIndicatorsForPopulationOptions
+});
+
+const getDefaults = createSelector(getFilterOptions, options => ({
+  popNationalIndicator: options.popNationalIndicator.find(
+    o => o.value === 'pop_total'
+  ),
+  popProvince: options.popProvince.find(o => o.value === 'Aceh')
+}));
+
+const getFieldSelected = field => state => {
+  const { query } = state.location;
+  if (!query || !query[field]) return getDefaults(state)[field];
+  const queryValue = query[field];
+  const options = getFilterOptions(state)[field];
+
+  return options.find(o => o.value === queryValue);
+};
+
+const getSelectedOptions = createStructuredSelector({
+  popNationalIndicator: getFieldSelected('popNationalIndicator'),
+  popProvince: getFieldSelected('popProvince')
+});
+
+const getFirstChartFilter = (queryName, selectedOptions) => {
+  const label = selectedOptions[queryName] && selectedOptions[queryName].label;
+
+  return [ { label } ];
+};
+
+const getBarChartData = createSelector(
+  [ getIndicators, getNationalIndicatorsForPopulation, getSelectedOptions ],
+  (data, indicators, selectedOptions) => {
+    if (!indicators || !data) return null;
+
+    const queryName = 'popNationalIndicator';
+
+    const selectedIndicator = indicators.find(
+      ind => ind.indicator_code === selectedOptions[queryName].value
+    );
+
+    const code = selectedIndicator && selectedIndicator.indicator_code;
+    const indicator = data &&
+      data.indicators &&
+      data.indicators.find(ind => ind.code === code);
+    const unit = indicator && indicator.unit;
+
+    const selectedData = [];
+    selectedIndicator.values.forEach(d => {
+      if (d.value && d.year) {
+        selectedData.push({ x: d.year, y: d.value });
+      }
+    });
+
+    return {
+      data: selectedData,
+      domain: { x: [ 'auto', 'auto' ], y: [ 0, 'auto' ] },
+      config: {
+        axes: {
+          xBottom: { name: 'Years', unit: '', format: 'string' },
+          yLeft: { name: 'Number of people', unit: '', format: 'number' }
+        },
+        tooltip: {
+          y: {
+            label: 'People',
+            format: value => `${format(',.2r')(`${value * 1000}`)}`
```

Comment:
Could we have this 1000 in a variable? To understand that is scaling because of the units